### PR TITLE
only run integration tests for a subset of integrations during PR builds

### PR DESCRIPTION
**Pull request by cgardens**
## What
* We won't always be able to run all integrations as part of every PR builds. This picks a subset of integrations to run for PR builds and then lets all tests run on master builds. This is not in place of some of the incrementality and caching fixes we want to do in the future but in addition. Ultimately we will need to do both.
* MVP for splitting as described [here](https://github.com/airbytehq/airbyte/issues/621).

### CODE REVIEW

File: tools/bin/integration_test_pr.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+assert_root
+
+./gradlew --no-daemon --scan -x generateProtocolClassFiles \
+  :airbyte-integrations:connectors:destination-bigquery:integrationTest \
+  :airbyte-integrations:connectors:destination-postgres:integrationTest \
+  :airbyte-integrations:connectors:destination-csv:integrationTest \
+  :airbyte-integrations:connectors:source-postgres-singer:integrationTest \
+  :airbyte-integrations:connectors:source-stripe-singer:integrationTest \
+  :airbyte-integrations:connectors:source-exchangeratesapi-singer:integrationTest
```
Comment (jrhizor) [inline tools/bin/integration_test_pr.sh:15]: nit: missing newline at end of file

File: tools/bin/standard_test_pr.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -e
+
+. tools/lib/lib.sh
+
+assert_root
+
+./gradlew --no-daemon --scan -x generateProtocolClassFiles \
+  :airbyte-integrations:connectors:source-github-singer:standardSourceTest
```
Comment (jrhizor) [inline tools/bin/standard_test_pr.sh:10]: nit: missing newline at end of file

File: .github/workflows/gradle.yml [hunk 3]
```diff
@@ -54,10 +54,20 @@ jobs:
...
+        if: success() && github.ref != 'refs/heads/master'
+        run: ./tools/bin/integration_test_pr.sh
+
+      - name: Run Integration Tests (Master)
+        if: success() && github.ref == 'refs/heads/master'
         run: ./gradlew --no-daemon integrationTest --scan -x generateProtocolClassFiles
 
-      - name: Run Standard Source Tests
+      - name: Run Standard Source Tests (PR)
+        if: success() && github.ref == 'refs/heads/master'
```
Comment (jrhizor) [inline .github/workflows/gradle.yml:66]: is the matching condition reversed?

File: .github/workflows/gradle.yml [hunk 4]
```diff
@@ -54,10 +54,20 @@ jobs:
...
+        run: ./tools/bin/standard_test_pr.sh
+
+      - name: Run Standard Source Tests (Master)
+        if: success() && github.ref != 'refs/heads/master'
```
Comment (jrhizor) [inline .github/workflows/gradle.yml:70]: also here

**Review by jrhizor — CHANGES_REQUESTED** [review]

**Comment by cgardens** [comment]
@jrhizor - nice catch. in fixing one those conditions being flipped, i managed to mess up the other. should be all good now.

**Review by jrhizor — APPROVED** [review]

**Pull request merged**